plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/OptInDialog.java

```diff
@@ -26,13 +28,28 @@ public OptInDialog(Shell parentShell) {
    */
   @Override
   protected Point getInitialLocation(Point initialSize) {
-    Rectangle parentBounds = getParentShell().getBounds();
-    Rectangle parentClientArea = getParentShell().getClientArea();
+    Shell targetShell = null;
+    try {
+      // Prefer showing at the top-right corner of the currently active workbench, but
+      // if we can't get a workbench for any reason, fall back to the parent shell.
+      IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
+      targetShell = window != null ? window.getShell() : getParentShell();
+    } catch (IllegalStateException ise) {  // getWorkbench() might throw this.
+      // No special handling. targetShell will be null.
+    }
 
-    int heightCaptionAndUpperBorder =
-        parentBounds.height - parentClientArea.height - getParentShell().getBorderWidth();
-    return new Point(parentBounds.x + parentClientArea.width - initialSize.x,
-        parentBounds.y + heightCaptionAndUpperBorder);
+    if (targetShell == null) {
+      return super.getInitialLocation(initialSize);
```

Comment:
The targetShell logic was not perfect anyway. I refactored this portion out, so hopefully it's more succinct.